maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/booterclient/ForkStarter.java

```diff
@@ -653,6 +653,12 @@ private RunResult fork( Object testSet, PropertiesWrapper providerProperties, Fo
             {
                 runResult = timeout( reporter.getGlobalRunStatistics().getRunResult() );
             }
+            else if ( forkClient.isErrorInFork() )
+            {
+                final StackTraceWriter stackTrace = forkClient.getErrorInFork();
+                booterForkException =
+                    new SurefireBooterForkException( stackTrace.writeTraceToString() );
+            }
```

Comment:
Please look line 711 - 733
StackTrace is printed if Maven command is started with -e option

Reply:
@fl4via
Pls have a look at the comments and answers. How can we continue with this?